quarkus/container/Dockerfile

```diff
@@ -25,6 +25,9 @@ ENV LANG en_US.UTF-8
 # Flag for determining app is running in container
 ENV KC_RUN_IN_CONTAINER true
 
+# Management interface is enabled by default for container
+ENV KC_MANAGEMENT_ENABLED true
```

Comment:
Thank you for describing this situation. I see the point for enabling health and management port by default when using containers and the KC Operator. 

Taking a step back, I assume the health was disabled by default only because we didn't have a separate management port at that time. If we would have had it back then, we would have enabled it by default. 
I also see it as a surprise to users that it behaves differently in a container when running standalone. I remember that Wildfly also opened multiple ports by default. 

So let's rethink it, and make the management port enabled in all environment by changing the default. No need for additional "hidden" ENV variables for the container. Users can still choose to disable it if they don't want it in either setup. 

Going one step further, I see no management port enabled and exposing health + metrics on the regular port as a bad practice and it should be deprecated. It could then be removed in a future release (it is hard to imagine a setup where you would actually want to do that). Once we ensure it is never active on the regular port, we could start enabling health (and possibly metrics) by default. 

TL;DR: Enabling the management port everywhere by default is IMHO the way to go for this PR. Everything else described above is out-of-scope here.

Reply:
> I also see it as a surprise to users that it behaves differently in a container when running standalone. I remember that Wildfly also opened multiple ports by default.

Yep, it'd be better to unify it. 

> So let's rethink it, and make the management port enabled in all environment by changing the default. No need for additional "hidden" ENV variables for the container. Users can still choose to disable it if they don't want it in either setup.

Sounds reasonable to me, thanks @ahus1!